PyDSS/reports/thermal_metrics.py

```diff
@@ -69,9 +69,61 @@ def generate(self, output_dir):
             "num_transformers": self._num_transformers,
             "max_violations": self._get_max_violations(),
             "moving_average_max_violations": self._get_moving_average_max_violations(),
+            "summary": {}
         }
+        
+        data['summary'] = self._sumarize_metrics(data)
 
         self._export_json_report(data, output_dir, "thermal_metrics.json")
+        
+    def _sumarize_metrics(self, data, scenario='control_mode'):
+        
+        transformer_window_dur_hrs = int(
+            self._transformer_window_size.total_seconds()/3600
+        )
+        line_window_dur_hrs = int(
+            self._line_window_size.total_seconds()/3600
+        )
+        
+        try:
+            line_dicts = data['max_violations'][scenario]['Lines']
+            max_line_inst_loading = (
+                max([x['max_violation'] for x in line_dicts])
+            )
+        except:
```

Comment:
Catching all possible exceptions is dangerous. A code bug in one of lines 89-91 could cause the except clause to run. It would be better to make your logic explicit. I'm guessing that this logic is conditional on something in `data`. Can you rewrite this as an if/else block with that conditional? You shouldn't need the try/except.

Same for the rest of the try blocks.

Reply:
Okay.